src/Menu.cpp

```
@@ -0,0 +1,84 @@
+#include "sudoManager/Menu.hpp"
+#include "Item.cpp"
+
+Menu::Menu() = default;
+
+Menu::Menu(const std::vector<Item>& vec)
+    : menu_items(vec) {}
+
+Menu::~Menu() = default;
+
+std::vector<Item>::iterator Menu::find_item(int item_ID) {
+    std::vector<Item>::iterator it = menu_items.begin();
+    while (it != menu_items.end() && (*it).getItemID() != item_ID) {
+        it++;
+    }
+    return it;
+}
+
+std::vector<Item>::iterator Menu::find_item(const std::string& item_name) {
+    std::vector<Item>::iterator it = menu_items.begin();
+    while (it != menu_items.end() && (*it).getName() != item_name) {
+        it++;
+    }
+    return it;
+}
+
+void Menu::add_menu_item(const Item& item) {
+    // adds item only if item doesn't exist in menu_items
+    // assuming item_ID to be a unique attribute
+    if (find_item(item.getItemID()) == menu_items.end()) {
+        menu_items.push_back(item);
+    } else {
+        //
+    }
+}
+
+void Menu::remove_menu_item(int item_ID) {
+    // erases item with ID: item_ID if exists
+    std::vector<Item>::iterator it = find_item(item_ID);
+    if (it != menu_items.end()) {
+        menu_items.erase(it);
+    }
+}
+
+void Menu::remove_menu_item(const std::string& item_name) {
+    // erases item with name: item_name if exists
+    std::vector<Item>::iterator it = find_item(item_name);
+    if (it != menu_items.end()) {
+        menu_items.erase(it);
+    }
+}
+
+Item Menu::get_menu_item(const std::string& item_name) {
+    std::vector<Item>::iterator it = find_item(item_name);
+    if (it == menu_items.end()) {
+        std::cout << "Item with Name: " << item_name << "doesn't exist !\n";
+        return Item(); 
+    } else {
+        return *it;
+    }
+}
+
+float Menu::get_menu_item_price(const std::string& item_name) {
+    std::vector<Item>::iterator it = find_item(item_name);
+    if (it == menu_items.end()) {
+        std::cout << "Item with Name: " << item_name << "doesn't exist !\n";
+        return -1.0; 
```

should throw an error instead.